rrsig.c

```diff
@@ -399,15 +402,15 @@ static void *rrsig_validate(struct rr *rrv)
 static pthread_mutex_t *lock_cs;
 static long *lock_count;
 
-static unsigned long pthreads_thread_id(void)
+unsigned long pthreads_thread_id(void)
 {
 	unsigned long ret;
 
 	ret=(unsigned long)pthread_self();
 	return(ret);
 }
 
-static void pthreads_locking_callback(int mode, int type, char *file, int line)
+void pthreads_locking_callback(int mode, int type, char *file, int line)
```

Comment:
The whole locking callbacks and memory allocation is not required for 1.1. I suggest you hide id somehwere.
Same goes for the library init which can be skipped as long as openssl ist used as a shared library.